### PR TITLE
Village and POI stuff

### LINKED ISSUE #2119: VillagerEntity#hasRecentlyWorkedAndSlept -> hasRecentlySlept
Likely changed in 20w19a. This method is used in golem spawning check.

### PR DESCRIPTION
**Pull request by apple502j**
Maps some function arguments, methods, and renames VillagerEntity#hasRecentlyWorkedAndSlept.

Fixes #2119

### CODE REVIEW

**Review by liach — APPROVED** [review]

**Comment by i509VCB** [comment]
cc @LambdAurora

**Comment by LambdAurora** [comment]
I see a lot of parameters mapping but few methods mapping, are those methods synthetic methods?
If not would be worth it and try to name them too.

And I personally really would appreciate Javadoc.

**Comment by liach** [comment]
most likely those are lambda params.

File: mappings/net/minecraft/world/poi/PointOfInterestType.mapping [hunk 1]
```diff
@@ -21,10 +21,16 @@ CLASS net/minecraft/class_4158 net/minecraft/world/poi/PointOfInterestType
...
+	METHOD method_19156 (Lnet/minecraft/class_4158;)Z
+		ARG 1 poiType
 	METHOD method_19161 getTicketCount ()I
+	METHOD method_19162 (Lnet/minecraft/class_4158;)Z
+		ARG 0 poiType
 	METHOD method_19164 getCompletionCondition ()Ljava/util/function/Predicate;
 	METHOD method_19516 from (Lnet/minecraft/class_2680;)Ljava/util/Optional;
 		ARG 0 state
+	METHOD method_19517 (Lnet/minecraft/class_2680;)Z
+		ARG 0 blockState
```
Comment (YanisBft) [inline mappings/net/minecraft/world/poi/PointOfInterestType.mapping]:
```suggestion
		ARG 0 state
```

**Review by YanisBft — APPROVED** [review]

**Review by enbrain — APPROVED** [review]

**Pull request merged**